src/main/java/frc/robot/subsystems/gyro/GyroIO.java

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+package frc.robot.subsystems.gyro;
+
+import org.littletonrobotics.junction.AutoLog;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+
+public interface GyroIO
+{
+    @AutoLog
+    public static class GyroIOInputs
+    {
+        public Rotation2d yawPosition          = new Rotation2d();
+        public double     yawVelocityRadPerSec = 0.0;
```

Comment:
We'll also want to add in a `Rotation2d` for the current tilt/roll. I see you made a method for that in the `GyroIONavX2` class, but the method for getting the roll/tilt should be in the subsystem, while the IO implementation class simply gives the roll back to the subsystem through the `GyroIOInputs` class.

```suggestion
        public Rotation2d yawPosition          = new Rotation2d();
        public double     yawVelocityRadPerSec = 0.0;
        public Rotation2d rollPosition         = new Rotation2d();
```